amethyst_tiles/src/iters.rs

```diff
@@ -0,0 +1,215 @@
+        (self.max.x - self.min.x) * (self.max.y - self.min.y) * ((self.max.z - self.min.z) + 1)
+    }
+
+    /// Create a linear iterator across this region.
+    pub fn iter(&self) -> RegionLinearIter {
+        RegionLinearIter::new(*self)
+    }
+}
+
+impl<T: AsRef<MortonRegion>> From<T> for Region {
+    fn from(region: T) -> Self {
+        let region = region.as_ref();
+        let min = morton::decode(region.min);
+        let max = morton::decode(region.max);
+
+        Self {
+            min: Point3::new(min.0, min.1, min.2),
+            max: Point3::new(max.0, max.1, max.2),
+        }
+    }
+}
+
+/// Linear iterator across a 3D coordinate space.
+pub struct RegionLinearIter {
+    track: Point3<u32>,
+    region: Region,
+}
+impl RegionLinearIter {
+    /// Create a new iterator.
+    pub fn new(region: Region) -> Self {
+        Self {
+            region,
+            track: region.min,
+        }
+    }
+}
+impl Iterator for RegionLinearIter {
+    type Item = Point3<u32>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let ret = self.track;
+
+        if self.track.z > self.region.max.z {
+            return None;
+        }
+
+        if self.track.x >= self.region.max.x {
+            self.track.y += 1;
+            self.track.x = self.region.min.x;
+        } else {
+            self.track.x += 1;
+            return Some(ret);
+        }
+
+        if self.track.y >= self.region.max.y {
+            self.track.z += 1;
+
+            self.track.y = self.region.min.y;
+        }
+
+        Some(ret)
+    }
+}
+
+#[cfg(test)]
+#[allow(clippy::shadow_unrelated)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn region_morton_match() {
+        let region = Region::new(Point3::new(0, 0, 0), Point3::new(64, 64, 64));
+        let morton_region = MortonRegion::new(
+            morton::encode(region.min.x, region.min.y, region.min.z),
+            morton::encode(region.max.x, region.max.y, region.max.z),
+        );
+        let morton_region_from = region.into();
+
+        assert_eq!(morton_region, morton_region_from);
+
+        let d = morton::decode(morton_region.min);
+        assert_eq!(region.min.x, d.0);
+        assert_eq!(region.min.y, d.1);
+        assert_eq!(region.min.z, d.2);
+
+        let d = morton::decode(morton_region.max);
+        assert_eq!(region.max.x, d.0);
+        assert_eq!(region.max.y, d.1);
+        assert_eq!(region.max.z, d.2);
+
+        let c = Point3::new(5, 5, 5);
+        assert_eq!(
+            morton_region.contains(morton::encode(c.x, c.y, c.z)),
+            region.contains(&c)
+        );
+
+        let c = Point3::new(11, 62, 0);
+        assert_eq!(
+            morton_region.contains(morton::encode(c.x, c.y, c.z)),
+            region.contains(&c)
+        );
+
+        let c = Point3::new(128, 55, 565);
+        assert_eq!(
+            morton_region.contains(morton::encode(c.x, c.y, c.z)),
+            region.contains(&c)
+        );
+    }
```

Comment:
there should probably be tests that check for not containing a point

Reply:
There are, there are tests included which check that both MortonRegion and Region return false together.

```        let c = Point3::new(128, 55, 565);
        assert_eq!(
            morton_region.contains(morton::encode(c.x, c.y, c.z)),
            region.contains(&c)
        );
```

Reply:
Oh, i see. This is very not clear, i'd prefer to have the `assert_eq` written directly against `true` and `false` values. That would be just easier to read. Also the tests don't check any boundary values.

Reply:
Boundaries were added in a new set of tests.